### PR TITLE
Add tons of docs

### PR DESCRIPTION
**Pull request by emptyflash**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,10 +1,43 @@
...
+ * `heimdall.host` (required) - routes a request only if its host matches this value
+ * `heimdall.path` (required) - path prefix the request must match (the prefix is stripped from the forwarded request)
+ * `heimdall.filters` - a JSON list of plugs through which to filter the request before sending it back off
+ * `heimdall.opts` - a JSON object of options that gets passed into each plug
+
+Filters can be viewed as a pipeline of plugs (simple functions that take a 
+request and return a changed request or a response), 
+with the final plug forwarding the request to the place it actually needs
+ to go. The pipeline is built in the order that the plugs are listed in the
+config. In other words, if `heimdal.filters` is: `["Plugs.First", "Plugs.Second"]`
```
Comment (kentongray) [inline README.md]: this is cool!

File: README.md [hunk 2]
```diff
@@ -22,3 +55,40 @@ If [available in Hex](https://hex.pm/docs/publish), the package can be installed
...
+      def init(opts), do: opts
+
+      def call(conn, _opts) do
+        conn
+        |> put_req_header("some-header", "a header value")
+      end
+    end
+    ```
+
+  4. Add heimdall labels to your marathon app config
```
Comment (kentongray) [inline README.md]: nit picky but should this be capitalized?

File: README.md [hunk 3]
```diff
@@ -1,10 +1,43 @@
...
+config. In other words, if `heimdal.filters` is: `["Plugs.First", "Plugs.Second"]`
+ the pipeline would have the following flow:
+
+`Request -> Plugs.First -> Plugs.Second -> Heimdall.Plug.ForwardRequest -> Microservice`
+
+Ideally, Heimdall should forward requests to a load balancer that knows how to
+send each request to an actual instance of the microservice it's trying to get
+to. Some recommended solutions for this are [traefik](https://github.com/containous/traefik), [Minuteman](https://github.com/dcos/minuteman), or your own
+custom nginx config. The default forward url (ie the location of the load
+balancer) is an OTP application setting, `:default_forward_url`. If you
```
Comment (kentongray) [inline README.md]: for us erlang newbs maybe a link to an example of how to set otp settings would be nice

File: lib/heimdall.ex [hunk 4]
```diff
@@ -1,2 +1,8 @@
 defmodule Heimdall do
+  @moduledoc Regex.replace(
+    ~r/```(elixir|json)(\n|.*)```/Us, 
+    File.read!("README.md"),
```
Comment (kentongray) [inline lib/heimdall.ex]: does this blank out code for the readme? maybe a comment could be nice

File: lib/dynamic_routes.ex [hunk 5]
```diff
@@ -1,22 +1,57 @@
...
 
   def init([tab: tab]), do: tab
 
+  @doc """
+  Returns the route in given routes that matches a path as a list (which
+  is how plug conns reperesent them internally).
+  
+  ## Examples
+
+      iex> Heimdall.DynamicRoutes.lookup_path([{"localhost", ["test", "path"], [], {}}], ["test", "path"])
```
Comment (kentongray) [inline lib/dynamic_routes.ex]: does this actually get run in the tests?

File: README.md [hunk 6]
```diff
@@ -1,10 +1,43 @@
...
+ * `heimdall.opts` - a JSON object of options that gets passed into each plug
+
+Filters can be viewed as a pipeline of plugs (simple functions that take a 
+request and return a changed request or a response), 
+with the final plug forwarding the request to the place it actually needs
+ to go. The pipeline is built in the order that the plugs are listed in the
+config. In other words, if `heimdal.filters` is: `["Plugs.First", "Plugs.Second"]`
+ the pipeline would have the following flow:
+
+`Request -> Plugs.First -> Plugs.Second -> Heimdall.Plug.ForwardRequest -> Microservice`
```
Comment (kentongray) [inline README.md]: do filters stream?

File: README.md (hunk 2 above)
Comment (emptyflash) [inline README.md]: Yep, I was trying to keep that consistent.

File: README.md (hunk 6 above)
Comment (emptyflash) [inline README.md]: For the most part they are all discrete functions that take in a "Conn" and return a "Conn". The Conn is an abstraction over the connection, and contains info about the request and the response. The only streaming that happens is if you read the Conn's request body or write to the Conn's response body.

File: lib/marathon/register_callback.ex [hunk 7]
```diff
@@ -24,6 +29,14 @@ defmodule Heimdall.Marathon.RegisterCallback do
     end
   end
 
+  @docs """
+  Sends a post requeset to Marathon's event subscriptions with the
```
Comment (kentongray) [inline lib/marathon/register_callback.ex]: typo!

**Pull request merged**